File: 1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const val2idx = new Map();

    for (let i = 0; i < nums.length; ++i) {
        const remains = target - nums[i];
        if (val2idx.get(remains) !== undefined) {
            return [i, val2idx.get(remains)];
        } else {
            val2idx.set(nums[i], i);
        }
    }
};
// @lc code=end

File: 1.两数之和.c
/*
 * @lc app=leetcode.cn id=1 lang=c
 *
 * [1] 两数之和
 */

// @lc code=start


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    int indexes[2] = [0, 1];

    for (; indexes[0] < numsSize - 1; ++indexes[0]) {
        for (indexes[1] = indexes[0] + 1; indexes[1] < numsSize; ++indexes[1]) {
            if (nums[indexes[0]] + nums[indexes[1]] == target) {
                return indexes;
            }
        }
    }
    return indexes;
}
// @lc code=end

